### ISSUE TITLE
NaN while training unconditional GAN with FFHQ by sample code in README

### ISSUE THREAD

**Issue by veryverypro** [lucidrains/gigagan-pytorch#60]
```
gan = GigaGAN(
    generator = dict(
        dim_capacity = 8,
        style_network = dict(
            dim = 64,
            depth = 4
        ),
        image_size = 256,
        dim_max = 512,
        num_skip_layers_excite = 4,
        unconditional = True
    ),
    discriminator = dict(
        dim_capacity = 16,
        dim_max = 512,
        image_size = 256,
        num_skip_layers_excite = 4,
        unconditional = True
    ),
    amp = True
).cuda()

# dataset

dataset = ImageDataset(
    folder = '/data/datasets/FFHQ/',
    image_size = 256
)
```

The problem seems fixed when I change amp to False. How can I fix this while still using amp = True ?

**Comment by veryverypro** [comment]
```
    scheduler = None
    if warmup_steps > 0:
        assert total_steps > 0, "total_steps must be provided if warmup_steps > 0"

        def lr_lambda(current_step):
            if current_step < warmup_steps:
                return current_step / warmup_steps
            return max(0.0, (total_steps - current_step) / (total_steps - warmup_steps))

        scheduler = LambdaLR(optimizer, lr_lambda)
```

Learning rate warmup will do the trick.
